src/actions/Bluetooth/SensorDownloadActions.js

```diff
@@ -98,26 +95,31 @@ const downloadLogsFromSensor = sensor => async dispatch => {
 
         if (downloadedLogsResult) {
           try {
-            const savedTemperatureLogs = UIDatabase.objects(VACCINE_ENTITIES.TEMPERATURE_LOG)
-              .filtered('sensor.macAddress == $0', macAddress)
-              .sorted('timestamp', true);
-
-            const [mostRecentLog] = savedTemperatureLogs;
-            const mostRecentLogTime = mostRecentLog ? mostRecentLog.timestamp : null;
-            const nextPossibleLogTime = mostRecentLogTime
-              ? moment(mostRecentLogTime).add(logInterval, 's')
-              : 0;
+            const mostRecentLogTime = moment(mostRecentLog ? mostRecentLog.timestamp : 0);
+            // This is the time which must have passed before downloading a new log.
+            // Either the time has passed the programmed date (ensuring no time travel),
+            // the logging delay must be in the past and the most recent log + the logging interval
+            // must be in the past.
+            const downloadBoundary = moment
+              .max([
+                moment(programmedDate),
+                moment(logDelay),
+                moment(mostRecentLogTime).add(logInterval, 's'),
+              ])
+              .unix();
+
+            const nextTimestamp = mostRecentLog ? moment.max([mostRecentLogTime, logDelay]) : null;
```

Comment:
Do `mostRecentLogTime` and `logDelay` need to be moment-ified here? Added a new sensor and been getting problems with getting the logs to load as is 🙀

Reply:
Ah thanks, sure does: a61a50e

Buuut.. I also just realised that it was quite right, anyway? So I tried to fix it with: 
539e903

Reply:
aaaand officially a mess f9d226d